internal/crawler/crawler.go

```diff
@@ -184,7 +185,12 @@ func (c *Crawler) visit(ctx context.Context, t Target) ([]*url.URL, Resource, er
 
 	// http error
 	if res.StatusCode >= 300 {
-		return nil, r, fmt.Errorf("%s response", res.Status)
+		shouldIgnore := c.Allow404 && res.StatusCode == 404
+		err := fmt.Errorf("%s response", res.Status)
```

Comment:
LGTM! The only thing I would change is to move the `if res.StatusCode = 404 && c.Allow404` above 187 since it checks StatusCode already we don't need to nest it in the branch

Reply:
@tj updated!

Reply:
perfect thanks!